pipelines/review-list/src/review_list/pipelines/review_list/nodes.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from functools import reduce
+
+import pyspark.sql as ps
+from pyspark.sql import Window
+from pyspark.sql.functions import col, row_number
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: maybe use inject_object()
+def combine_ranked_pair_dataframes(
+    weights: dict[str, dict], config: dict[str, any], **dataframes: ps.DataFrame
+) -> ps.DataFrame:
+    """
+    Combine ranked pair dataframes using weighted merge logic.
+
+    Args:
+        weights: Parameter structure with weights configuration
+        config: Configuration parameters including limit
+        **dataframes: Individual dataframes passed as keyword arguments
+    """
+    if not dataframes:
+        raise ValueError("At least one DataFrame must be provided")
+
+    limit = config["limit"]
+    weight_values = {}
+    for dataset_name, weight_config in weights.items():
+        # TODO: handle all edge cases
+        if "weight" not in weight_config:
+            raise ValueError(f"Missing weight for dataset {dataset_name}")
+        weight_values[dataset_name] = weight_config["weight"]
+
+    dfs_with_weights = [
+        (dataframes[name], weight_values[name]) for name in dataframes.keys()
+    ]
+
+    return weighted_merge_multiple(dfs_with_weights, limit)
+
+
+def weighted_merge_multiple(dfs_with_weights, limit):
+    """
+    Merge any number of Spark DataFrames according to weights and rank order.
+
+    Args:
+        dfs_with_weights (list): List of tuples [(df1, weight1), (df2, weight2), ...]
```

Comment:
```suggestion
        dfs_with_weights: List of tuples [(df1, weight1), (df2, weight2), ...]
```
nitpick but suggest to keep to docstring format that we've (generally) been adhereing to in MATRIX